# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.h
# Commit: b6e830c59764
# Full Hash: b6e830c59764b9d175faf177642ab8a9d26c8139
# Author: stransky <stransky@redhat.com>
# Date: 2022-09-19 15:58:06
# Regressor Bug: 1789581
# File Overlap Count: 1
# Description:
#   Bug 1789581 [Wayland] Use new popup position at WaylandPopupFitsToplevelWindow() if popup is moved r=emilio
#   
#   If popup is moved we can't use recent popup position (mBounds) to check popup placement but
#   use requested position given at nsWindow::NativeMoveResize().
#   
# ==============================================================================

diff -r 37b28252e060 -r b6e830c59764 widget/gtk/nsWindow.h
--- a/widget/gtk/nsWindow.h	Mon Sep 19 10:39:15 2022 +0000
+++ b/widget/gtk/nsWindow.h	Mon Sep 19 10:39:16 2022 +0000
@@ -856,7 +856,7 @@
   void WaylandPopupMarkAsClosed();
   void WaylandPopupRemoveClosedPopups();
   void WaylandPopupSetDirectPosition();
-  bool WaylandPopupFitsToplevelWindow();
+  bool WaylandPopupFitsToplevelWindow(bool aMove);
   const WaylandPopupMoveToRectParams WaylandPopupGetPositionFromLayout();
   void WaylandPopupPropagateChangesToLayout(bool aMove, bool aResize);
   nsWindow* WaylandPopupFindLast(nsWindow* aPopup);
